### PR TITLE
Loading messages for channel

### PR DESCRIPTION
**Pull request by siepra**
…t of messages ids. Prepared store for keeping ids and its corresponding messages

### CODE REVIEW

File: src/components/Message/Message.test.tsx [hunk 1]
```diff
@@ -8,174 +8,15 @@ describe('Message component', () => {
...
-                verticalTextAlign="center"
-              >
-                Brownie powder marshmallow dessert carrot cake marzipan cake caramels. Muffin topping wafer jelly apple pie candy. Fruitcake chocolate pudding fruitcake candy lemon drops chocolate.
-              </Text>
-            </View>
-          </View>
-        </View>
-      </View>
-    `);
+    expect(toJSON()).toMatchInlineSnapshot();
```
Comment (kowalski) [inline src/components/Message/Message.test.tsx:20]: that's a bid odd that this tests passes. I think it means that the component is returning `null`. Perhaps the props that you pass to it are incorrect?

File: src/store/publicChannels/publicChannels.selectors.ts [hunk 2]
```diff
@@ -55,9 +30,55 @@ export const ZbayChannel = createSelector(
...
+
+export const currentChannelMessages = createSelector(
+  selectReducer(StoreKeys.PublicChannels),
+  reducerState => reducerState.currentChannelMessages,
+);
+
+export const missingCurrentChannelMessages = createSelector(
+  currentChannelMessagesIds,
+  currentChannelMessages,
+  (ids, messages) => {
```
Comment (kowalski) [inline src/store/publicChannels/publicChannels.selectors.ts:50]: putting this more concisely:
```
(ids, messages) => ids.filter(id => !(id in messages))
```

File: src/store/publicChannels/publicChannels.selectors.ts [hunk 3]
```diff
@@ -55,9 +30,55 @@ export const ZbayChannel = createSelector(
...
+        missing.push(id);
+      }
+    });
+    return missing;
+  },
+);
+
+export const currentChannelDisplayableMessages = createSelector(
+  currentChannelMessages,
+  messages => {
```
Comment (kowalski) [inline src/store/publicChannels/publicChannels.selectors.ts:63]:
```
messages => Object.entries(messages)).map(([id, message]) => ({
        id: id,
        message: message.message,
        nickname: 'anon',
        datetime: formatMessageDisplayDate(message.createdAt),
      }))
      ```

File: src/store/publicChannels/publicChannels.slice.ts [hunk 4]
```diff
@@ -3,54 +3,78 @@ import { createSlice, EntityState, PayloadAction } from '@reduxjs/toolkit';
...
+import { IChannelInfo, IMessage } from './publicChannels.types';
 
 export class PublicChannelsState {
-  public currentChannelAddress: string = '';
   public channels: EntityState<IChannelInfo> =
     publicChannelsAdapter.getInitialState();
-  public messages: EntityState<IMessage> =
-    publicChannelMessagesAdapter.getInitialState();
+
+  public currentChannelMessagesIds: EntityState<string> =
```
Comment (kowalski) [inline src/store/publicChannels/publicChannels.slice.ts:15]: Well, I know that in your app you only have one channel, but what is the plan when you have more than one? 
I think we need a structure like:

```
{ [channelId: string]: { ids: string[], data: [id: string]: IMessage } }
```
This way you track state of multiple channels.

"currentChannelMessagesIds" becomes a selector, that draws from currentChannelId key

File: src/store/publicChannels/publicChannels.slice.ts [hunk 5]
```diff
@@ -3,54 +3,78 @@ import { createSlice, EntityState, PayloadAction } from '@reduxjs/toolkit';
...
         state.channels,
         Object.values(action.payload),
       );
     },
     subscribeForTopic: (state, _action: PayloadAction<IChannelInfo>) => state,
+    responseSendIds: (
+      state,
+      action: PayloadAction<ChannelMessagesIdsResponse>,
+    ) => {
+      publicChannelMessagesIdsAdapter.setAll(
```
Comment (kowalski) [inline src/store/publicChannels/publicChannels.slice.ts:63]: I would not use entityAdapter for messageIds. 
You only ever replace all of them. I think I would use a simple `string[]` for that

**Review by kowalski — CHANGES_REQUESTED** [review]

File: src/components/Message/Message.test.tsx (hunk 1 above)
Comment (siepra) [reply src/components/Message/Message.test.tsx:20]: I've cleared this one as the params has changed, and I did not run it since then

**Review by kowalski — APPROVED** [review]

**Review by kowalski — APPROVED** [review]

**Pull request merged**